subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionReport.kt

```diff
@@ -119,45 +162,52 @@ class InstantExecutionReport(
 
     private
     fun errors() =
-        problems.filterIsInstance<PropertyProblem.Error>()
+        problems.toList().filterIsInstance<PropertyProblem.Error>()
 
     private
-    fun logSummary() {
-        logger.warn(summary())
+    fun logSummary(outputDirectory: File) {
+        logger.warn(summary(outputDirectory))
     }
 
     private
-    fun writeReportFiles() {
-        outputDirectory.mkdirs()
-        copyReportResources()
-        writeJsReportData()
+    fun writeReportFiles(outputDirectory: File) {
+        require(outputDirectory.deleteRecursively()) {
+            "Could not clean instant execution report directory '$outputDirectory'"
+        }
+        require(outputDirectory.mkdirs()) {
+            "Could not create instant execution report directory '$outputDirectory'"
+        }
+        copyReportResources(outputDirectory)
+        writeJsReportData(outputDirectory)
     }
 
     private
-    fun summary(): String {
-        val uniquePropertyProblems = problems.groupBy {
-            propertyDescriptionFor(it) to it.message
-        }
+    fun summary(outputDirectory: File): String {
+        val uniquePropertyProblems = problems
+            .sortedBy { it.trace.sequence.toList().reversed().joinToString(".") }
```

Comment:
:x: why sorting here?

And by the way, no test fails if I remove that line.

Reply:
It was remaining from some experiment I did. Removed.

Reply:
Reinstated on `master`, see https://github.com/gradle/gradle/commit/0ecd45121feb435a994b956b8114806afd7aa67d